daemon/device-plugin/deviceplugin.go

```diff
@@ -104,45 +115,40 @@ func (nf *nfResources) Allocate(ctx context.Context, rqt *pluginapi.AllocateRequ
 	return resp, nil
 }
 
-func (nf *nfResources) GetDeviceState(DeviceName string) string {
-	// TODO: Discover device health
-	return pluginapi.Healthy
-}
-
-func (nf *nfResources) Start() error {
-	nf.cleanup()
-	nf.ensureConnected()
-
-	ctx := context.Background()
+func (nf *nfResources) GetDevices() error {
+	err := nf.ensureConnected()
+	if err != nil {
+		return err
+	}
 
-	Devices, err := nf.client.GetDevices(ctx, &pb.Empty{})
+	Devices, err := nf.client.GetDevices(context.Background(), &pb.Empty{})
 	if err != nil {
 		nf.log.Error(err, "Failed to handle GetDevices Request")
 		return err
 	}
 
 	for _, device := range Devices.Devices {
 		nf.devices[device.ID] = pluginapi.Device{ID: device.ID, Health: pluginapi.Healthy}
+		nf.log.Info("Found device", "device.ID", device.ID)
 	}
 
-	for dev := range nf.devices {
-		nf.log.Info(dev)
-	}
+	return nil
+}
 
+func (nf *nfResources) RegisterDevicePlugin() error {
 	pluginEndpoint := filepath.Join(pluginapi.DevicePluginPath, nf.socketFile)
-	fmt.Printf("Starting NF Device Plugin server at: %s\n", pluginEndpoint)
+	nf.log.Info("Starting Device Plugin server at:", "pluginEndpoint", pluginEndpoint)
 	lis, err := net.Listen("unix", pluginEndpoint)
 	if err != nil {
-		fmt.Printf("Error: Starting NF Device Plugin server failed: %v", err)
+		nf.log.Error(err, "Failed to listen to Device Plugin server at:", "resourceName", resourceName, "pluginEndpoint", pluginEndpoint)
+		return err
 	}
 	nf.grpcServer = grpc.NewServer()
 
 	kubeletEndpoint := filepath.Join("unix:", DeprecatedSockDir, KubeEndPoint)
-
 	conn, err := grpc.Dial(kubeletEndpoint, grpc.WithTransportCredentials(insecure.NewCredentials()))
-
 	if err != nil {
-		fmt.Printf("%s device plugin unable connect to Kubelet : %v", resourceName, err)
+		nf.log.Error(err, "Device Plugin unable connect to Kubelet:", "resourceName", resourceName, "kubeletEndpoint", kubeletEndpoint)
```

Comment:
Since we will return from `Start` here, we should not log the error here. We will log it at the `Start`-call site. Same remark for the rest of the PR.

Reply:
Done